### ISSUE TITLE
creatures with zero power not considered for attack

### ISSUE THREAD

**Issue by melvinzhang** [magarena/magarena#759]
muppet @ http://www.slightlymagic.net/forum/viewtopic.php?f=82&t=17210&p=196524#p196524

This one is more of an attack with ai hints.

If you give it say 20 0/2 creatures and a Signal Pest I am pretty sure it will still fail to attack and kill you.

So I'm not sure what is wrong with the attacking code but it doesn't seem to understand 0 power creatures can be powered up.

It might be that it simply doesn't even bother calculating whether to attack at all as there is nothing in the log in the attack step. So maybe it needs a don't skip the attack step think even if the only creatures are 0 power line.

**Comment by melvinzhang** [comment]
In MagicDeclareAttackersResultBuilder.java, creatures with power <= 0 are removed from consideration as potential attackers. But they may have their power increased later due to an attack trigger or spell.

**Comment by melvinzhang** [comment]
One fix is to remove this heuristic completely, but that may result in pointless attacks when there is no possibility of pumping the creatures with <= 0 power.

**Comment by ShawnieBoy** [comment]
These 'pointless' attacks won't always appear that way, as the player will not be able to (normally) see the AI's hand. Only if the attack can be blocked and the creature destroyed would it be seen as pointless.

During it's simulations, does the AI compare the battlefield before and after combat to determine if attacking it worth it? Even just a permanent count difference? That could help in determining whether to attack with 0-power creatures.

**Comment by melvinzhang** [comment]
Yes, after blockers are determined, the top 12 blocks per attack are selected base on the result of combat resolution. See MagicGameCombatScore.java
